hooks/src/index.js

```diff
@@ -184,7 +184,23 @@ export function useReducer(reducer, initialState, init) {
 
 		if (!currentComponent._hasScuFromHooks) {
 			currentComponent._hasScuFromHooks = true;
-			const prevScu = currentComponent.shouldComponentUpdate;
+			let prevScu = currentComponent.shouldComponentUpdate;
+			const prevCWU = currentComponent.componentWillUpdate;
+
+			// If we're dealing with a forced update `shouldComponentUpdate` will
+			// not be called. But we use that to update the hook values, so we
+			// need to call it.
+			currentComponent.componentWillUpdate = function(p, s, c) {
```

Comment:
Is there any risk that this happens over and over for the same component, leading to more and more wrapping?

Reply:
This branch is guarded by a check a couple of lines earlier which ensures that this code is only executed once per component.

```js
if (!currentComponent._hasScuFromHooks) {
  currentComponent._hasScuFromHooks = true;
  // ...
}
```